### PR TITLE
Fix build on Node 7

### LINKED ISSUE #636: Unable to build apm under Node v7.0.0
**Description**:
When attempting to build Atom from source, the process fails out when installing apm. Probably related to #635 since I can build just fine if I revert to node v6.9.1.

**Steps to Reproduce:**
1. Have node v7.0.0 installed
2. Fresh clone of atom
3. Run `script/build`

**Stacktrace**:

```
child_process.js:145
    throw new TypeError('Incorrect value of args option');
    ^

TypeError: Incorrect value of args option
    at Object.exports.execFile (child_process.js:145:11)
    at Object.exports.exec (child_process.js:108:18)
    at Object.<anonymous> (/home/dnoel/src/atom/apm/node_modules/atom-package-manager/script/postinstall.js:10:16)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.runMain (module.js:607:10)
    at run (bootstrap_node.js:382:7)
npm ERR! Linux 4.8.6-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "--global-style" "--loglevel=error" "install"
npm ERR! node v7.0.0
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
    at checkExecSyncError (child_process.js:488:13)
    at Object.execFileSync (child_process.js:508:13)
    at module.exports (/home/dnoel/src/atom/script/lib/install-apm.js:10:16)
    at Object.<anonymous> (/home/dnoel/src/atom/script/bootstrap:27:1)
    at Module._compile (module.js:573:32)
    at Object.Module._extensions..js (module.js:582:10)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)
    at Module.require (module.js:500:17)
```

### PR DESCRIPTION
**Pull request by lgeiger**
Use `child_process.spawn` instead of  `child_process.exec`.

Fixes #636

### CODE REVIEW

**Comment by lgeiger** [comment]
We could also keep using `exec` but then the second argument must be removed since it is invalid.

**Comment by drewmnoel** [comment]
Been using this fork today as part of my normal workflow (list installed, add new, remove existing), LGTM.

**Comment by lgeiger** [comment]
@jsolisu I think this is a bug in the code that is just hidden because node < 7.0.0 didn't throw a error.

Looking at the [API docs of Node.js 6.9.1](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_child_process_exec_command_options_callback) the function signature should be
```javascript
child_process.exec(command[, options][, callback])
```

What's used in the code is the signature of spawn
```javascript
child_process.spawn(command[, args][, options])
```

I'm surprised it worked before 😄 .

**Comment by drewmnoel** [comment]
Been using it just fine, as expected. Can we get a core member to do some testing too so we can get this merged in?

**Comment by underoot** [comment]
Hm, it's works early, because `exec` execute `execFile` with `args = []` and `options = { stdio: ['pipe', 'pipe', 'pipe'] }` and without `callback`. But after nodejs/node@32037b78a20ba9df2e80cae7d2c0f4ce9efbeac9 `execFile` has check of `callback` existence. 

Using `spawn` looks right now.

**Comment by mpkh** [comment]
I can confirm that without it atom can't be compiled with node7, so +1 from me there

File: script/postinstall.js [hunk 1]
```diff
@@ -7,6 +7,6 @@ if (process.platform.indexOf('win') === 0) {
 } else {
   script += '.sh'
 }
-var child = cp.exec(script, [], {stdio: ['pipe', 'pipe', 'pipe']})
+var child = cp.spawn(script, [], {stdio: ['pipe', 'pipe', 'pipe']})
```
Comment (thomasjo) [inline script/postinstall.js]: This change won't work as expected on Windows. You need to add `shell: true` to the options object;
```js
var child = cp.spawn(script, [], { stdio: ['pipe', 'pipe', 'pipe'], shell: true })
```

---
_Ref: https://nodejs.org/dist/latest-v7.x/docs/api/child_process.html#child_process_spawning_bat_and_cmd_files_on_windows_

Comment (lgeiger) [inline script/postinstall.js]: Fixed 👍

**Comment by thomasjo** [comment]
@atom/feedback This LGTM. Let's ship a beta release, so we can fully test this in conjunction with a PR on atom/atom?

**Comment by lgeiger** [comment]
@thomasjo Thanks for the review. Could we get #638 in the beta release as well?

**Comment by ronjouch** [comment]
Great, was discussing this with the maintainer of the [AUR/atom-editor-beta package](https://aur.archlinux.org/packages/atom-editor-beta/), which currently builds against node 6.9.1. We'll submit feedback if the move to 7 doesn't go well.

File: .travis.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ node_js:
   - "0.10"
```
Comment (joefitzgerald) [inline .travis.yml]: Remove `0.10`; apm no longer uses this version

File: .travis.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ node_js:
...
   - "4"
```
Comment (joefitzgerald) [inline .travis.yml]: Remove `4`; apm no longer uses this version

**Comment by BinaryMuse** [comment]
Let's keep v4 in the build matrix; it's not there because apm used it, it's there to make sure apm can be built on a machine with Node 4. Removing 0.10 is fine.

**Comment by handicraftsman** [comment]
#MergeIt

**Pull request merged**